### PR TITLE
wallet-rpc: get transfers for all accounts and subaddresses

### PR DESCRIPTION
**Pull request by jtgrassie**
#5109 for get_transfers

### CODE REVIEW

File: src/wallet/wallet_rpc_server.cpp [hunk 1]
```diff
@@ -31,6 +31,7 @@
 #include <boost/asio/ip/address.hpp>
 #include <boost/filesystem/operations.hpp>
 #include <boost/algorithm/string.hpp>
+#include <boost/optional/optional.hpp>
```
Comment (stoffu) [inline src/wallet/wallet_rpc_server.cpp]: This is unnecessary as `boost::optional` is already used in `wallet2.h`.

Comment (jtgrassie) [reply src/wallet/wallet_rpc_server.cpp]: Thanks. I missed that.

File: src/net/CMakeLists.txt [hunk 2]
```diff
@@ -30,5 +30,5 @@ set(net_sources error.cpp parse.cpp socks.cpp tor_address.cpp)
 set(net_headers error.h parse.h socks.h tor_address.h)
 
 monero_add_library(net ${net_sources} ${net_headers})
-target_link_libraries(net epee ${Boost_ASIO_LIBRARY})
+target_link_libraries(net common epee ${Boost_ASIO_LIBRARY})
```
Comment (stoffu) [inline src/net/CMakeLists.txt]: This is already covered by #5140

Comment (jtgrassie) [reply src/net/CMakeLists.txt]: Yes. But for the sake of such a small change that's unmerged, I kept this so builds would succeed.

**Pull request merged**